[Feat] react-router-dom v6 & styled-components 추가 및 간단한 페이지 및 컴포넌트 추가

dmstmdrbs opened the pull request:
- styled-components와 styled-reset을 추가했어요!
  - GlobalStyles를 이용해 reset CSS를 적용했어요
- react-router-dom v6를 이용해 라우터를 구성했어요
  - Main 페이지와 Create 페이지를 구성했어요
- Header 컴포넌트를 추가하여 현재 페이지 정보를 나타낼 수 있도록 했어요

dmstmdrbs committed:
chore: add styled-components & reset CSS with styled-reset

dmstmdrbs committed:
feat: add Routes for Main page & Create page and add Header components

메인페이지와 스터디 생성 페이지 라우터 추가
헤더 컴포넌트 추가

dmstmdrbs committed:
chore: add eslint rules related with single quote

single quote관련 ESLint 룰 추가
single quote를 우선으로 적용

Coreight98 commented on src/components/Header/index.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const Header = () => {
+  const location = useLocation();
+  const currentTitle = useMemo(() => {
```
지금 currentTitle은 단순히 현재의 타이틀만 가지고 있는 함수보다는 링크를 통한 타이틀을 반환하는 함수라고 생각되는데, 맞을까요?
이 경우에는 경로에 따른 반환을 처리하는 함수를 따로 분리하시고, title을 따로 넣는게 더 좋아 보여요.
추후에 경로가 추가 될 수 있어서 별도로 분리하는 부분도 좋다고 생각해요.
지금 딱히 생각나는 네이밍이 없는데, 아니면 TITLE_PATH 로 두고, object로 만들어서 as const 를 붙여서 상수화 해주시는 부분도 좋을거 같다고 생각되요!
개인적으로 생각했던 부분이니까, 좋은 방법 있으시면 말씀 부탁드려요!

dmstmdrbs replied:
이번 프로젝트에서는 링크가 많지 않을 것 같아서 오브젝트로 간단하게 만드는 것도 좋을 것 같네요!

```typescript
  const TITLE_FROM_PATH = {
  '/' : '스터디 목록',
  '/create' : '스터디 생성',
 } as const;
```

이런 식으로 하는 건 괜찮을까요?

Coreight98 replied:
고민이 되는게, 경로를 key로 두는것이 맞을까 라는 고민이 들어요
그래서 
``` typescript
'list' : {
 path : '/',
name : '스터디 목록'
}
```
으로 경로와 네임을 가진 객체로 두는 건 어떤가요 ?? 객체상의 목적은 이게 맞을거 같은데, 가져다 쓰는 함수를 하나 더 만들어야 하는 불편함은 존재할 것이라고 생각되네요.

dmstmdrbs replied:
Header 컴포넌트를 지우고 페이지 별로 적용할 수 있는 Layout컴포넌트를 추가했습니다! 헤더 타이틀을 따로 관리하지 않아도 적용할 수 있도록 구현했습니다

Coreight98 commented on src/components/Header/index.tsx:
```diff
@@ -0,0 +1,34 @@
+    <NavContainer>
+      {location.pathname !== '/' && (
+        <ButtonLink to='/'>
+          <GoBackButton>&larr;</GoBackButton>
```
태그에는 동작에 대한 네이밍보다는 단순 backButton이 더 적합하다고 생각해요!

dmstmdrbs replied:
좋습니다! 반영하겠습니다

Coreight98 reviewed (commented):
고생많으셨습니다!

dmstmdrbs committed:
feat: add Layout component & delete Header Component

각 페이지 별 적용할 수 있는 레이아웃 컴포넌트를 추가했어요!
  - props를 통해 옵션을 설정할 수 있어요
	- 추후 BottomNav을 추가할 수 있어요

dmstmdrbs committed:
style: modify max-width property of Layout
